src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -170,9 +175,56 @@ export class MicrobitSimulator extends React.Component<any, IState> {
         event.preventDefault();
         this.handleButtonClick(key, true);
     };
-
     protected onMouseLeave = (event: Event, key: string) => {
         event.preventDefault();
         console.log(`To implement onMouseLeave ${key}`);
     };
+    protected onKeyEvent(event: KeyboardEvent, active: boolean, key: string) {
+        event.stopPropagation();
+        if ([event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.ENTER)) {
+            this.handleButtonClick(key, active);
+            if (this.imageRef.current) {
+                if (key === BUTTONS_KEYS.BTN_A) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_A,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_B) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_B,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_AB) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_AB,
+                        active
+                    );
+                }
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.A)
+        ) {
+            this.handleButtonClick(BUTTONS_KEYS.BTN_A, active);
+            if (this.imageRef.current) {
+                this.imageRef.current.updateButtonAttributes(
+                    BUTTONS_KEYS.BTN_A,
+                    active
+                );
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.B)

```

Comment:
Perhaps consider adding a keyboard button to press both a and b simultaneously.

Reply:
done with C :)